subsys/debug/Kconfig

```diff
@@ -84,7 +84,7 @@ config EARLY_CONSOLE
 config ASSERT
 	bool
 	prompt "Enable __ASSERT() macro"
-	default n
+	default y if TEST
```

Comment:
Just `default TEST` should work here too. Might be inconsistent with the style elsewhere though.